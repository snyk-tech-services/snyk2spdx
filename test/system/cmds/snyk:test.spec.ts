import { exec } from 'child_process';
import * as path from 'path';
import { deleteFiles } from '../../delete-files';

const main = './dist/index.js'.replace(/\//g, path.sep);

describe('`snyk2spdx snyk:test`', () => {
  const filesToDelete: string[] = [];
  afterEach(async () => {
    deleteFiles(filesToDelete);
  });
  it('Shows help text as expected', (done) => {
    exec(`node ${main} help`, (err, stdout) => {
      if (err) {
        throw err;
      }
      expect(err).toBeNull();
      expect(stdout.trim()).toMatchSnapshot();
      done();
    });
  }, 70000);
  it('Shows help text as expected when calling command explicitly', (done) => {
    exec(`node ${main} snyk:jest help`, (err, stdout) => {
      if (err) {
        throw err;
      }
      expect(err).toBeNull();
      expect(stdout.trim()).toMatchSnapshot();
      done();
    });
  }, 70000);
});
