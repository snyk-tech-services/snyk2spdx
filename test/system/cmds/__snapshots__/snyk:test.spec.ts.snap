// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk2spdx snyk:test\` Shows help text as expected 1`] = `
"index.js

Convert \`snyk test --json\` output to SPDX SBOM

Commands:
  index.js snyk:test  Convert \`snyk test --json\` output to SPDX SBOM   [default]

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
  --output   Save the output to the specified file name. Defaults to stdout"
`;

exports[`\`snyk2spdx snyk:test\` Shows help text as expected when calling command explicitly 1`] = `
"index.js

Convert \`snyk test --json\` output to SPDX SBOM

Commands:
  index.js snyk:test  Convert \`snyk test --json\` output to SPDX SBOM   [default]

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
  --output   Save the output to the specified file name. Defaults to stdout"
`;
